### ISSUE TITLE
[SAE] 'softmax_topk' torch path and closed-form .fit() optimize different models

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#601]
**Verified — CONFIRMED.**

The torch sparsity layer implements **independent** nonnegative top-k via softplus magnitudes + hard top-k STE (`_topk_gate`, `gamfit/torch/manifold_sae.py:507-544`), with an explicit comment that softmax is wrong for SAE activations (it can't turn all atoms off). But the closed-form assignment mapper coerces it to plain softmax:
```python
"softmax_topk": "softmax",   # gamfit/torch/manifold_sae.py:285
```
and `.fit()` (~821-838) ships that to Rust, which fits `AssignmentMode::softmax(tau)` (`crates/gam-pyffi/src/lib.rs` ~10215). PyFFI treats top_k as a *post-hoc* hard projection after smooth softmax fitting (~10008-10017).

These are different objectives. With all logits = -100: independent top-k → softplus(-100)≈0 (reconstruction ≈ off); softmax → 1/K each (must use an atom). Mapping `softmax_topk → softmax` violates the module's own parity contract.

**Fix:** add a true `AssignmentMode::IndependentTopK { k, temperature, magnitude }` in Rust (u_k = softplus(logit/τ)·τ, mask = top-k(u), a_k = u_k·mask, STE or sparsemax/entmax surrogate). Map `"softmax_topk" → "independent_topk"`. Consider BatchTopK (batch-level constraint) too. Do not use row-softmax unless the model is explicitly a mixture assignment.

_Source: static SAE/manifold-SAE audit; verified against current code at commit 3c1a4efa2._

**Comment by SauersML** [comment]
Resolved by design. `softmax_topk` is no longer mapped to competitive softmax — it now maps to `jumprelu` (manifold_sae.py:337-341), the independent magnitude-carrying gate that, unlike softmax, lets all atoms turn off. The original bug (independent activations coerced to a simplex) is gone; a separate `IndependentTopK` Rust mode was judged redundant given jumprelu already provides independent gating. Documented at manifold_sae.py:324-335.
